Data-Structures/src/main/java/graph/Graph.java

```diff
@@ -67,4 +67,38 @@ public Graph() {
     public int size() {
         return this.nodes.size();
     }
+
+    // accept a starting node and perform a breadth-first traversal on the graph
+    // return a collection of the nodes in the order they were visited
+    public String breadthFirstTraversal(Node start) {
+        HashSet<Node<T>> visited = new HashSet<>();
+        Queue<Node<T>> toVisit = new Queue<>();
+        LinkedList<Node<T>> ans = new LinkedList<>();
+        // StringJoiner is for testing
+        StringJoiner result = new StringJoiner(", ");
+
+        visited.add(start);
+        ans.add(start);
+        toVisit.enqueue(start);
+        result.add(start.getValue().toString());
+        while (!toVisit.isEmpty()) {
+            Node<T> current = toVisit.peek();
+            for (Edge<T> neighbor: current.getNeighbors()) {
+                Node neighborNode = neighbor.getNode();
+                if (!visited.contains(neighborNode)) {
+                    toVisit.enqueue(neighborNode);
+                    ans.add(neighborNode);
+                    visited.add(neighborNode);
+                    // for testing
+                    result.add(neighborNode.getValue().toString());
+                    continue;
+                }
+            }
+            toVisit.dequeue();
+        }
+//        return ans;
+
+        // For testing purpose, return a string that shows the nodes in order that they were visited
+        return result.toString();

```

Comment:
I understand this makes testing easier, but I would be in favor of returning the linked list `ans` in this method and then doing the string joiner stuff in the test file if you have to.